nucypher/blockchain/eth/actors.py

```diff
@@ -917,7 +930,7 @@ def initialize_stake(self,
         elif not entire_balance and not amount:
             raise ValueError("Specify an amount or entire balance, got neither")
 
-        token_balance = self.token_balance
+        token_balance = self.token_balance if not only_lock else self.calculate_staking_reward()
```

Comment:
Is this logic local to this block - or can it be part of `self.token_balance`?

Reply:
it's local logic